lib/web/userpreferences.go

```diff
@@ -74,17 +75,9 @@ func (h *Handler) updateUserPreferences(_ http.ResponseWriter, r *http.Request,
 	}
 
 	preferences := &userpreferencesv1.UpsertUserPreferencesRequest{
-		Username: sctx.GetUser(),
-		Preferences: &userpreferencesv1.UserPreferences{
-			Theme: req.Theme,
-			Assist: &userpreferencesv1.AssistUserPreferences{
-				PreferredLogins: req.Assist.PreferredLogins,
-				ViewMode:        req.Assist.ViewMode,
-			},
-			Onboard: &userpreferencesv1.OnboardUserPreferences{
-				PreferredResources: req.Onboard.PreferredResources,
-			},
-		},
+		Username:    sctx.GetUser(),
+		Preferences: userPreferencesResponseToProto(req),
+		Fieldmask:   fieldmask.FromStruct(req),
```

Comment:
You can build the fieldmask directly from the json representation similar to what grpc gateway 
With that, you can correctly patch each field without having to assume users won't replace fields with their zero values - for now, it might not be a problem mas in the future it can be